test/denormalized_has_one_test.rb

```diff
@@ -120,4 +120,45 @@ def test_cache_has_one_on_derived_model_raises
       StiRecordTypeA.cache_has_one :polymorphic_record, :inverse_name => :owner, :embed => true
     end
   end
+
+  def test_returned_record_should_be_readonly_on_cache_hit
+    IdentityCache.with_fetch_read_only_records do
+      Item.fetch_by_title('foo')
```

Comment:
Why not `Item.fetch(1)` here? That'd make it clearer that the subsequent `fetch` is a cache hit.